{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:28.873540Z",
     "start_time": "2018-10-08T17:18:26.653340Z"
    }
   },
   "outputs": [],
   "source": [
    "# Put these at the top of every notebook, to get automatic reloading and inline plotting\n",
    "%reload_ext autoreload\n",
    "%autoreload 2\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:32.093466Z",
     "start_time": "2018-10-08T17:18:28.875525Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import datetime\n",
    "\n",
    "import gtfs_utils\n",
    "\n",
    "sns.set_style(\"white\")\n",
    "sns.set_context(\"talk\")\n",
    "sns.set_palette('Set2', 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:32.307880Z",
     "start_time": "2018-10-08T17:18:32.096444Z"
    }
   },
   "outputs": [],
   "source": [
    "FOLDER = 'data\\\\siri\\\\2018-08'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:34.022078Z",
     "start_time": "2018-10-08T17:18:32.310871Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>agency</th>\n",
       "      <th>bus_id</th>\n",
       "      <th>end_time</th>\n",
       "      <th>line_num</th>\n",
       "      <th>route_id</th>\n",
       "      <th>service_id</th>\n",
       "      <th>start_time</th>\n",
       "      <th>time_recorded</th>\n",
       "      <th>lat</th>\n",
       "      <th>lon</th>\n",
       "      <th>date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>7591</td>\n",
       "      <td>660</td>\n",
       "      <td>19</td>\n",
       "      <td>2340</td>\n",
       "      <td>33441985</td>\n",
       "      <td>84300</td>\n",
       "      <td>86348</td>\n",
       "      <td>32.035385</td>\n",
       "      <td>34.767807</td>\n",
       "      <td>2018-08-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>7666</td>\n",
       "      <td>2340</td>\n",
       "      <td>19</td>\n",
       "      <td>2340</td>\n",
       "      <td>33643429</td>\n",
       "      <td>0</td>\n",
       "      <td>86109</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2018-08-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>10196</td>\n",
       "      <td>300</td>\n",
       "      <td>16</td>\n",
       "      <td>2323</td>\n",
       "      <td>33425076</td>\n",
       "      <td>84000</td>\n",
       "      <td>86372</td>\n",
       "      <td>32.045105</td>\n",
       "      <td>34.809189</td>\n",
       "      <td>2018-08-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>7853</td>\n",
       "      <td>1200</td>\n",
       "      <td>16</td>\n",
       "      <td>2323</td>\n",
       "      <td>33425077</td>\n",
       "      <td>85200</td>\n",
       "      <td>86380</td>\n",
       "      <td>32.055714</td>\n",
       "      <td>34.781860</td>\n",
       "      <td>2018-08-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>10159</td>\n",
       "      <td>2100</td>\n",
       "      <td>16</td>\n",
       "      <td>2323</td>\n",
       "      <td>33425078</td>\n",
       "      <td>0</td>\n",
       "      <td>86109</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2018-08-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   agency bus_id  end_time line_num  route_id  service_id  start_time  \\\n",
       "0       5   7591       660       19      2340    33441985       84300   \n",
       "1       5   7666      2340       19      2340    33643429           0   \n",
       "2       5  10196       300       16      2323    33425076       84000   \n",
       "3       5   7853      1200       16      2323    33425077       85200   \n",
       "4       5  10159      2100       16      2323    33425078           0   \n",
       "\n",
       "   time_recorded        lat        lon       date  \n",
       "0          86348  32.035385  34.767807 2018-08-01  \n",
       "1          86109   0.000000   0.000000 2018-08-01  \n",
       "2          86372  32.045105  34.809189 2018-08-01  \n",
       "3          86380  32.055714  34.781860 2018-08-01  \n",
       "4          86109   0.000000   0.000000 2018-08-01  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os.path\n",
    "s = pd.read_parquet(os.path.join(FOLDER, 'siri_rt_data.2018-08-01.0.log.parq'))\n",
    "s.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:36.942542Z",
     "start_time": "2018-10-08T17:18:34.025006Z"
    }
   },
   "outputs": [],
   "source": [
    "from dask.distributed import Client"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:47.194949Z",
     "start_time": "2018-10-08T17:18:36.951523Z"
    }
   },
   "outputs": [],
   "source": [
    "#>>> client = Client('127.0.0.1:8786')\n",
    "client = Client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:18:54.867719Z",
     "start_time": "2018-10-08T17:18:47.206918Z"
    }
   },
   "outputs": [],
   "source": [
    "import dask.dataframe as dd\n",
    "\n",
    "df = dd.read_parquet(FOLDER+'\\\\*.parq', index='service_id', engine='pyarrow')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:34:56.206431Z",
     "start_time": "2018-10-08T17:34:53.079795Z"
    }
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "rs = (pd.concat([pd.read_pickle(f, compression='gzip') \n",
    "                 for f in glob.glob('data\\\\gtfs_stats_MOD_ffill\\\\2018-08-*_route_stats.pkl.gz')])\n",
    "      .assign(date = lambda x: pd.to_datetime(x.date))\n",
    "      .assign(dayofweek = lambda x: x['date'].dt.strftime('%A')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:23:12.373426Z",
     "start_time": "2018-10-08T17:23:11.961528Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['9024', '11179', '11178', '21722', '20750'], dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs[(rs.route_short_name=='111') & (rs.route_long_name.str.contains('מודיעין'))].route_id.value_counts().index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:34:35.265432Z",
     "start_time": "2018-10-08T17:34:34.981193Z"
    }
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'Series' object has no attribute 'route_id'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-21-534b01572def>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mrs\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mroute_id\u001b[0m\u001b[1;33m==\u001b[0m\u001b[1;34m'16561'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m__getattr__\u001b[1;34m(self, name)\u001b[0m\n\u001b[0;32m   4374\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_can_hold_identifiers_and_holds_name\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4375\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4376\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4377\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4378\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'Series' object has no attribute 'route_id'"
     ]
    }
   ],
   "source": [
    "rs[(rs.route_id=='16561')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-18T12:48:09.550507Z",
     "start_time": "2018-09-18T12:48:09.316187Z"
    }
   },
   "outputs": [],
   "source": [
    "#df.memory_usage().sum().compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-18T12:48:09.769198Z",
     "start_time": "2018-09-18T12:48:09.550507Z"
    }
   },
   "outputs": [],
   "source": [
    "#len(df) # 153997865"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:26:56.870453Z",
     "start_time": "2018-10-08T17:26:56.327905Z"
    }
   },
   "outputs": [],
   "source": [
    "r = df[(df.lat!=0) & (df.route_id.isin([9024, 11179, 11178, 21722, 20750]))]# & (df.date>'2018-08-29')]#.drop_duplicates(subset=['bus_id', 'route_id', 'date', 'time_recorded'])\n",
    "#r = client.persist(r)\n",
    "r = r.repartition(npartitions=105)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:28:19.099555Z",
     "start_time": "2018-10-08T17:26:57.828893Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "rs = r.groupby('route_id').size().compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-08T17:28:31.332124Z",
     "start_time": "2018-10-08T17:28:30.673453Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "route_id\n",
       "9024      9911\n",
       "11178    59347\n",
       "11179    47483\n",
       "20750     4632\n",
       "21722     2710\n",
       "dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-18T12:34:17.370502Z",
     "start_time": "2018-09-18T12:34:17.167411Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "route_id\n",
       "13428    359789\n",
       "13429    357670\n",
       "2533     277098\n",
       "11685    247464\n",
       "11689    245487\n",
       "2532     243754\n",
       "9811     191053\n",
       "9813     185735\n",
       "2567     175139\n",
       "9803     169403\n",
       "2457     168274\n",
       "2566     164167\n",
       "10797    161534\n",
       "19740    160983\n",
       "9802     160285\n",
       "2454     155175\n",
       "10804    155166\n",
       "2328     146721\n",
       "13435    143505\n",
       "9816     140539\n",
       "9808     139498\n",
       "2332     137884\n",
       "9817     137816\n",
       "9821     135744\n",
       "9822     135706\n",
       "10799    134319\n",
       "2510     134119\n",
       "2492     133727\n",
       "9809     132483\n",
       "2961     131014\n",
       "          ...  \n",
       "4685         41\n",
       "17442        41\n",
       "16909        41\n",
       "2890         40\n",
       "11415        39\n",
       "17993        38\n",
       "23448        37\n",
       "7837         36\n",
       "8846         34\n",
       "10465        34\n",
       "11401        33\n",
       "425          32\n",
       "8110         32\n",
       "23449        32\n",
       "16908        31\n",
       "426          27\n",
       "19377        26\n",
       "12653        24\n",
       "11412        23\n",
       "19153        23\n",
       "23447        20\n",
       "11414        17\n",
       "4312         16\n",
       "16058        15\n",
       "11410        12\n",
       "16009        11\n",
       "11411        11\n",
       "17170         9\n",
       "11110         7\n",
       "13599         5\n",
       "Length: 6194, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs.sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-18T12:34:20.322923Z",
     "start_time": "2018-09-18T12:34:20.104255Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "67832523"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs.sum() # 67 million without dedup, only remove zero coords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-18T12:48:22.500603Z",
     "start_time": "2018-09-18T12:48:22.281891Z"
    }
   },
   "outputs": [],
   "source": [
    "def chunk(s):\n",
    "    '''\n",
    "    The function applied to the\n",
    "    individual partition (map)\n",
    "    '''    \n",
    "    return s.apply(lambda x: list(set(x)))\n",
    "\n",
    "\n",
    "def agg(s):\n",
    "    '''\n",
    "    The function whic will aggrgate \n",
    "    the result from all the partitions(reduce)\n",
    "    '''\n",
    "    s = s._selected_obj    \n",
    "    return s.groupby(level=list(range(s.index.nlevels))).sum()\n",
    "\n",
    "\n",
    "def finalize(s):\n",
    "    '''\n",
    "    The optional functional that will be \n",
    "    applied to the result of the agg_tu functions\n",
    "    '''\n",
    "    return s.apply(lambda x: len(set(x)))\n",
    "\n",
    "\n",
    "tunique = dd.Aggregation('tunique', chunk, agg,finalize)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-09-18T12:54:46.332530Z",
     "start_time": "2018-09-18T12:48:23.875795Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.nanny - WARNING - Worker exceeded 95% memory budget. Restarting\n",
      "distributed.nanny - WARNING - Worker process 7964 was killed by signal 15\n",
      "distributed.nanny - WARNING - Restarting worker\n",
      "distributed.nanny - WARNING - Worker exceeded 95% memory budget. Restarting\n",
      "distributed.nanny - WARNING - Worker process 5280 was killed by signal 15\n",
      "distributed.nanny - WARNING - Restarting worker\n",
      "tornado.application - ERROR - Uncaught exception GET /status/ws?bokeh-protocol-version=1.0&bokeh-session-id=ricUmtV9gbFx3AG4MXglJPa1lhDrrry1DD91Ufp1sDkc (127.0.0.1)\n",
      "HTTPServerRequest(protocol='http', host='localhost:8787', method='GET', uri='/status/ws?bokeh-protocol-version=1.0&bokeh-session-id=ricUmtV9gbFx3AG4MXglJPa1lhDrrry1DD91Ufp1sDkc', version='HTTP/1.1', remote_ip='127.0.0.1', headers={'Host': 'localhost:8787', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:62.0) Gecko/20100101 Firefox/62.0', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'Accept-Language': 'en-US,en;q=0.7,he;q=0.3', 'Accept-Encoding': 'gzip, deflate', 'Sec-Websocket-Version': '13', 'Origin': 'http://localhost:8787', 'Sec-Websocket-Extensions': 'permessage-deflate', 'Sec-Websocket-Key': 'MdmCJHWai/J8hoeOGLW3qg==', 'Cookie': 'username-localhost-8888=\"2|1:0|10:1537272997|23:username-localhost-8888|44:NTU5YWI3OTkzMDk4NDQxZmI1MzY3MTM5NzIwNjYwZTY=|926651cfd1f23480d2b71f962ba7b475033061f04e3bab469619f7a375f26a04\"; username-localhost-8889=\"2|1:0|10:1535023670|23:username-localhost-8889|44:OWE2MDAwNjRmNzljNDBlZmIzMzYwNzA4NWIzM2E0ZDk=|53846531d7270d0763dd3fe0a37af88733143df4579b751b2d936ab60338e9ad\"; _xsrf=2|4e84eec7|829da5a87a44b7dd8f1638eef34ae990|1536179415', 'Connection': 'keep-alive, Upgrade', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'Upgrade': 'websocket'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 185, in on_message\n",
      "    yield self._schedule(work)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 260, in _schedule\n",
      "    yield self.send_message(work)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 241, in send\n",
      "    yield conn.write_message(self.content_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 406, in write\n",
      "    self._handle_write()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 872, in _handle_write\n",
      "    del self._write_buffer[:self._write_buffer_pos]\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\web.py\", line 1467, in _stack_context_handle_exception\n",
      "    raise_exc_info((type, value, traceback))\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\stack_context.py\", line 316, in wrapped\n",
      "    ret = fn(*args, **kwargs)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 502, in <lambda>\n",
      "    self.stream.io_loop.add_future(result, lambda f: f.result())\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 188, in on_message\n",
      "    self._internal_error(\"server failed to handle a message\")\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 268, in _internal_error\n",
      "    self.close(10000, message)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 355, in close\n",
      "    self.ws_connection.close(code, reason)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 975, in close\n",
      "    self._write_frame(True, 0x8, close_data)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849B30A58> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849CF18D0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EDAD68> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EECF60> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849B836A0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298483AC278> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x000002984946E240> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298493DA470> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849DBB908> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849DBBC18> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849E963C8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x000002984941AEB8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EA2F60> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EB6080> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849ECF4A8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EECBE0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F01128> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F01A90> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EDAD30> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EDA550> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849DBBCF8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298494C50F0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849ECF0B8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EDAF60> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EB2BA8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849E96F60> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849E96940> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849D9AE10> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298493DAF60> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298475A2E10> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849D8A278> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849CF1C18> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849AD7550> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849CF1358> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F1B588> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F25E80> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849E99438> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849C85898> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849D73780> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EECA20> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F1B5C0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849D73FD0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298475A2588> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849D8ADA0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x00000298483ACF60> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x000002984837DCF8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849D9A4A8> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849E967F0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EA2B38> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EB2828> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EB6198> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849ECF630> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849EDA9B0> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F44A90> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F4A080> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n",
      "tornado.application - ERROR - Future <tornado.concurrent.Future object at 0x0000029849F5E978> exception was never retrieved: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 211, in send_message\n",
      "    yield message.send(self)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1063, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\protocol\\message.py\", line 229, in send\n",
      "    yield conn.write_message(self.header_json, locked=False)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 1055, in run\n",
      "    value = future.result()\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\concurrent.py\", line 238, in result\n",
      "    raise_exc_info(self._exc_info)\n",
      "  File \"<string>\", line 4, in raise_exc_info\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 307, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\bokeh\\server\\views\\ws.py\", line 227, in write_message\n",
      "    yield super(WSHandler, self).write_message(message, binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 252, in write_message\n",
      "    return self.ws_connection.write_message(message, binary=binary)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 785, in write_message\n",
      "    return self._write_frame(True, opcode, message, flags=flags)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\websocket.py\", line 768, in _write_frame\n",
      "    return self.stream.write(frame)\n",
      "  File \"C:\\Users\\dbare\\Anaconda3\\lib\\site-packages\\tornado\\iostream.py\", line 395, in write\n",
      "    self._write_buffer += data\n",
      "BufferError: Existing exports of data: object cannot be re-sized\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-13-7a9041e44906>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0md\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'date'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'route_id'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0magg\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'time_recorded'\u001b[0m \u001b[1;33m:\u001b[0m \u001b[0mtunique\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcompute\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\dask\\base.py\u001b[0m in \u001b[0;36mcompute\u001b[1;34m(self, **kwargs)\u001b[0m\n\u001b[0;32m    154\u001b[0m         \u001b[0mdask\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbase\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcompute\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    155\u001b[0m         \"\"\"\n\u001b[1;32m--> 156\u001b[1;33m         \u001b[1;33m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcompute\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtraverse\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    157\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    158\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\dask\\base.py\u001b[0m in \u001b[0;36mcompute\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    393\u001b[0m     \u001b[0mkeys\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__dask_keys__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mcollections\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    394\u001b[0m     \u001b[0mpostcomputes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__dask_postcompute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mcollections\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 395\u001b[1;33m     \u001b[0mresults\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mschedule\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdsk\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkeys\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    396\u001b[0m     \u001b[1;32mreturn\u001b[0m \u001b[0mrepack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mf\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m*\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0ma\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mzip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpostcomputes\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    397\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\distributed\\client.py\u001b[0m in \u001b[0;36mget\u001b[1;34m(self, dsk, keys, restrictions, loose_restrictions, resources, sync, asynchronous, direct, retries, priority, fifo_timeout, **kwargs)\u001b[0m\n\u001b[0;32m   2216\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2217\u001b[0m                 results = self.gather(packed, asynchronous=asynchronous,\n\u001b[1;32m-> 2218\u001b[1;33m                                       direct=direct)\n\u001b[0m\u001b[0;32m   2219\u001b[0m             \u001b[1;32mfinally\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2220\u001b[0m                 \u001b[1;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mfutures\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\distributed\\client.py\u001b[0m in \u001b[0;36mgather\u001b[1;34m(self, futures, errors, maxsize, direct, asynchronous)\u001b[0m\n\u001b[0;32m   1579\u001b[0m             return self.sync(self._gather, futures, errors=errors,\n\u001b[0;32m   1580\u001b[0m                              \u001b[0mdirect\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdirect\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlocal_worker\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mlocal_worker\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1581\u001b[1;33m                              asynchronous=asynchronous)\n\u001b[0m\u001b[0;32m   1582\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1583\u001b[0m     \u001b[1;33m@\u001b[0m\u001b[0mgen\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcoroutine\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\distributed\\client.py\u001b[0m in \u001b[0;36msync\u001b[1;34m(self, func, *args, **kwargs)\u001b[0m\n\u001b[0;32m    645\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mfuture\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    646\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 647\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0msync\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloop\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    648\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    649\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__repr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\distributed\\utils.py\u001b[0m in \u001b[0;36msync\u001b[1;34m(loop, func, *args, **kwargs)\u001b[0m\n\u001b[0;32m    273\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    274\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_set\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 275\u001b[1;33m             \u001b[0me\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m10\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    276\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0merror\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    277\u001b[0m         \u001b[0msix\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreraise\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0merror\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\threading.py\u001b[0m in \u001b[0;36mwait\u001b[1;34m(self, timeout)\u001b[0m\n\u001b[0;32m    549\u001b[0m             \u001b[0msignaled\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_flag\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    550\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0msignaled\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 551\u001b[1;33m                 \u001b[0msignaled\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_cond\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    552\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0msignaled\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    553\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\threading.py\u001b[0m in \u001b[0;36mwait\u001b[1;34m(self, timeout)\u001b[0m\n\u001b[0;32m    297\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    298\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 299\u001b[1;33m                     \u001b[0mgotit\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mwaiter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    300\u001b[0m                 \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    301\u001b[0m                     \u001b[0mgotit\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mwaiter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "d = r.groupby(['date', 'route_id']).agg({'time_recorded' : tunique}).compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-09-18T01:52:13.488Z"
    }
   },
   "outputs": [],
   "source": [
    "c = r.groupby(['date', 'route_id']).size().compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-09-18T08:36:16.420Z"
    }
   },
   "outputs": [],
   "source": [
    "d.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
